From ea989a9d00b00e6e90f4d0e12e9637a64ffa3f9d Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Thu, 27 Aug 2020 14:27:29 +0800
Subject: [PATCH] Revert "drm: introduce atomic_begin for connector"

This reverts commit 401fc0060c12a476b12f94df21ad5af0e4506b6b.

For compatibility with GKI, connector atomic_begin/atomic_flush
should be removed.

Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
Change-Id: Ibff527242a2938027e0f7871a84e549513d5c2ea
---
 drivers/gpu/drm/drm_atomic_helper.c      | 22 ----------------------
 include/drm/drm_modeset_helper_vtables.h |  8 --------
 2 files changed, 30 deletions(-)

diff --git a/drivers/gpu/drm/drm_atomic_helper.c b/drivers/gpu/drm/drm_atomic_helper.c
index 9274ac77ed25..cf8af4dcfae0 100644
--- a/drivers/gpu/drm/drm_atomic_helper.c
+++ b/drivers/gpu/drm/drm_atomic_helper.c
@@ -2361,28 +2361,6 @@ void drm_atomic_helper_commit_planes(struct drm_device *dev,
 	bool active_only = flags & DRM_PLANE_COMMIT_ACTIVE_ONLY;
 	bool no_disable = flags & DRM_PLANE_COMMIT_NO_DISABLE_AFTER_MODESET;
 
-	for_each_oldnew_connector_in_state(old_state, connector,
-					   old_connector_state,
-					   new_connector_state, i) {
-		const struct drm_connector_helper_funcs *funcs;
-
-		if (!connector->state->crtc)
-			continue;
-
-		if (!connector->state->crtc->state->active)
-			continue;
-
-		funcs = connector->helper_private;
-
-		if (!funcs || !funcs->atomic_begin)
-			continue;
-
-		DRM_DEBUG_ATOMIC("flush beginning [CONNECTOR:%d:%s]\n",
-				 connector->base.id, connector->name);
-
-		funcs->atomic_begin(connector, old_connector_state);
-	}
-
 	for_each_oldnew_crtc_in_state(old_state, crtc, old_crtc_state, new_crtc_state, i) {
 		const struct drm_crtc_helper_funcs *funcs;
 
diff --git a/include/drm/drm_modeset_helper_vtables.h b/include/drm/drm_modeset_helper_vtables.h
index 886c6e9d8f28..a772dea3afbf 100644
--- a/include/drm/drm_modeset_helper_vtables.h
+++ b/include/drm/drm_modeset_helper_vtables.h
@@ -1001,14 +1001,6 @@ struct drm_connector_helper_funcs {
 	void (*atomic_commit)(struct drm_connector *connector,
 			      struct drm_connector_state *state);
 
-	/**
-	 * @atomic_begin:
-	 *
-	 * begin atomic update.
-	 */
-	void (*atomic_begin)(struct drm_connector *connector,
-			     struct drm_connector_state *conn_state);
-
 	/**
 	 * @atomic_flush:
 	 *
-- 
2.35.3

